storybook/package.json

```diff
@@ -6,11 +6,21 @@
     "build-storybook": "storybook build",
     "test": "test-storybook --stories-json",
     "chromatic": "chromatic --exit-zero-on-changes --project-token=b1bb684c98a2",
-    "format": "prettier --write src",
-    "lint:code": "eslint src --max-warnings=0",
-    "lint:formatting": "prettier --check src",
+    "get-changed": "git diff --relative -r --name-only --diff-filter=ACMRTUXB origin/main | grep -E '(.ts$|.tsx$|.js$|.jsx$)' | tr '\n' ' ' > ./.lint-cache/changes.txt",
+    "if-changed": "changed=$(cat ./.lint-cache/changes.txt) && (node -e \"if ('$changed' !== ''){process.exit(1)} else {console.log('No changes detected')}\" ||  \"${@}\")",
+    "if-get-changed": "yarn get-changed && yarn if-changed",
+    "format-all": "prettier --write src",
+    "format-pattern": "prettier --write --cache --cache-location './.lint-cache/.prettier-cache'",
+    "format": "yarn if-get-changed yarn format-pattern $(cat ./.lint-cache/changes.txt)",
+    "lint-all:code": "eslint src --max-warnings=0",
+    "lint-pattern:code": "eslint --max-warnings=0 --cache --cache-location './.lint-cache/' --cache-strategy content",
+    "lint:code": "yarn if-get-changed yarn lint-pattern:code $(cat ./.lint-cache/changes.txt)",
+    "lint-all:formatting": "prettier --check src",
+    "lint-pattern:formatting": "prettier --check",
+    "lint:formatting": "yarn if-get-changed yarn lint-pattern:formatting $(cat ./.lint-cache/changes.txt)",
     "lint:types": "tsc --noEmit",
-    "lint": "yarn lint:code && yarn lint:formatting && yarn lint:types"
+    "lint": "yarn get-changed && yarn if-changed yarn lint-pattern:formatting $(cat ./.lint-cache/changes.txt) & yarn if-changed yarn lint-pattern:code $(cat ./.lint-cache/changes.txt) & yarn lint:types",
+    "lint-all": "yarn lint-all:formatting & yarn lint-all:code & yarn lint:types"
```

Comment:
We should use [concurrently](https://github.com/open-cli-tools/concurrently) for running these commands